breakpad-symbols/src/lib.rs

```diff
@@ -569,6 +571,32 @@ fn module_key(module: &(dyn Module + Sync)) -> ModuleKey {
     )
 }
 
+struct CachedAsyncResult<T, E> {
+    inner: FutMutex<Option<Arc<Result<T, E>>>>,
+}
+
+impl<T, E> Default for CachedAsyncResult<T, E> {
+    fn default() -> Self {
+        CachedAsyncResult {
+            inner: FutMutex::new(None),
+        }
+    }
+}
+
+impl<T, E> CachedAsyncResult<T, E> {
+    pub async fn get<'a, F, Fut>(&self, f: F) -> Arc<Result<T, E>>
+    where
+        F: FnOnce() -> Fut + 'a,
+        Fut: std::future::Future<Output = Result<T, E>> + 'a,
+    {
+        let mut guard = self.inner.lock().await;
+        if guard.is_none() {
+            *guard = Some(Arc::new(f().await));
+        }
+        unsafe { guard.as_ref().unwrap_unchecked() }.clone()
```

Comment:
there's no way this unsafe is justified, just use normal unwrap. the compiler will probably optimize it out anyway, and it won't be a fun landmine for a refactor.

Reply:
Whoops, definitely not justified.